Fantasy Battle Game/Assets/Scripts/BattleManagement/TurnManagement.cs

```diff
@@ -0,0 +1,137 @@
+        }
+
+        public void Winner()
+        {
+            if (endGame_)
+            {
+                if (lastMoveFirstPlayer_) { Debug.Log("Win " + firstPlayer_.Name); }
+                else { Debug.Log("Win " + secondPlayer_.Name); }
+            }
+        }
+
+        public void StartBattle()
+        {
+            if (firstPlayer_== null || secondPlayer_ == null)
+            {
+                Debug.LogError("no first or second player");
+                return;
+            }
+
+            championsManager_ = ChampionsManager.Instance;
+            firstPlayerTurnList_ = new List<Champion>();
+            secondPlayerTurnList_ = new List<Champion>();
+        }
+
+        public bool NextTurn()
+        {
+            if (firstPlayer_.Champions.Count == 0 || secondPlayer_.Champions.Count == 0)
+            {
+                Winner();
+                return false;
+            }
+            if (lastMoveFirstPlayer_)
+            {
+                lastMoveFirstPlayer_ = false;
+                endGame_ = secondPlayerTurn();
+            }
+            else
+            {
+                lastMoveFirstPlayer_ = true;
+                endGame_ = firstPlayerTurn();
+            }
+            return !endGame_;
+        }
+
+        private bool firstPlayerTurn()
+        {
+            Champion selectedChampion;
+            do
+            {
+                if (firstPlayerTurnList_.Count == 0)
+                {
+                    if (firstPlayer_.Champions.Count == 0) { return false; }
+                    foreach (var champion in firstPlayer_.Champions)
+                    {
+                        firstPlayerTurnList_.Add(champion);
+                    }
+                }
+                selectedChampion = firstPlayerTurnList_[0];
+                firstPlayerTurnList_.RemoveAt(0);
+            } while (!firstPlayer_.Champions.Contains(selectedChampion));
+
+            Debug.Log("first player turn");
+            championsManager_.SelectedChampion = selectedChampion;
+            return true;
+        }
+
+        private bool secondPlayerTurn()
+        {
+            Champion selectedChampion;
+            do
+            {
+                if (secondPlayerTurnList_.Count == 0)
+                {
+                    if (secondPlayer_.Champions.Count == 0) { return false; }
+                    foreach (var champion in secondPlayer_.Champions)
+                    {
+                        secondPlayerTurnList_.Add(champion);
+                    }
+                }
+                selectedChampion = secondPlayerTurnList_[0];
+                secondPlayerTurnList_.RemoveAt(0);
+            } while (!secondPlayer_.Champions.Contains(selectedChampion));
+
+            Debug.Log("second player turn");
+            championsManager_.SelectedChampion = selectedChampion;
+            return true;
+        }
+
+        public TurnManagement()
+        {
+            if (Instance == null)
```

Comment:
Consider ternary statement as solution. It makes things easier, especially if you got really simple statement to control.
`instance_ = Instance == null ? this : null;`